mfe/mfe.go

```diff
@@ -0,0 +1,1357 @@
+package mfe
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"regexp"
+	"strings"
+)
+
+/******************************************************************************
+May, 26, 2021
+
+The MFE calculation functionality (including comments and description of
+	functions) has been taken directly from ViennaRNA with the follwing changes:
+- ViennaRNA includes the ability to specify a dangle model (more info available
+	at [src/bin/RNAeval.ggo#L153](https://github.com/ViennaRNA/ViennaRNA/blob/d6fbaf2b33d65944f9249a91ed5ab4b3277f7d06/src/bin/RNAeval.ggo#L153)).
+	This implementation keeps it simple and defaults to the value of -d2.
+- ViennaRNA includes the ability to specify hard and soft constraints (more info
+	available from [their ppt explaining hard and soft constrains](http://benasque.org/2015rna/talks_contr/2713_lorenz_benasque_2015.pdf),
+	[official docs](https://www.tbi.univie.ac.at/RNA/ViennaRNA/doc/html/group__constraints.html),
+	and [their paper](https://almob.biomedcentral.com/articles/10.1186/s13015-016-0070-z)).
+	This implementation keeps it simple and defaults to no hard or soft
+	constraints.
+- ViennaRNA includes the ability to calculate the minimum free energy of
+	co-folded sequences. This implementation keeps it simple and defaults to
+	calculating the mfe of only single sequences.
+
+File is structured as so:
+
+	Structs:
+		foldCompound - holds all information needed to compute the free energy of a
+			RNA secondary structure (a RNA sequence along with its folded structure).
+		EnergyContribution - holds information about the energy contribution of a
+			single loop, and the indexes of the base pairs that delimit the loop in
+			the RNA secondary structure.
+
+	Big functions from this file:
+
+		MinimumFreeEnergy - given a RNA sequence, it's structure, and a temperature,
+			it returns the minimum free energy of the RNA secondary structure at the
```

Comment:
Given a secondary structure and specified temperature, is the "free energy" not determined? If it is determined then: 
```suggestion
			it returns the free energy of the RNA secondary structure at the
```
If not, out of curiosity, what else factors into it?

Also, what is the difference between _free_ energy and just energy?